homeassistant/components/arve/entity.py

```diff
@@ -0,0 +1,39 @@
+"""Arve base entity."""
+
+from __future__ import annotations
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import ArveCoordinator
+
+
+class ArveDeviceEntity(CoordinatorEntity[ArveCoordinator]):
+    """Defines a base Arve device entity."""
+
+    _attr_has_entity_name = True
+    _attr_available = True
+
+    def __init__(
+        self, coordinator: ArveCoordinator, description: EntityDescription
+    ) -> None:
+        """Initialize the Arve device entity."""
+        super().__init__(coordinator)
+
+        self.entity_description = description
+
+        self._attr_unique_id = (
+            f"{coordinator.arve.device_sn}_{self.entity_description.translation_key}"
```

Comment:
```suggestion
            f"{coordinator.arve.device_sn}_{description.translation_key}"
```
I see using name or translation key in the unique id as a bad practice as we don't use this often (so people think they can change it without consequences), I'd rather see it using the `key`